### PR TITLE
feat(router): add payments authentication api flow

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add payments authentication api flow

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Compiler guided and sanity payments flows testing

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/openapi/src/routes/payments.rs [hunk 1]
```diff
@@ -450,3 +450,23 @@ pub fn payments_list() {}
...
+  params(
+      ("payment_id" = String, Path, description = "The identifier for payment")
+  ),
+  responses(
+      (status = 200, description = "Authentication created", body = PaymentsExternalAuthenticationResponse),
+      (status = 400, description = "Missing mandatory fields")
+  ),
+  tag = "Payments",
+  operation_id = "Initiate external authentication for a Payment",
+  security(("api_key" = []))
```
Comment (Narayanbhat166) [inline crates/openapi/src/routes/payments.rs:470]:
```suggestion
  security(("publishable_key" = []))
```

File: crates/router/src/core/payments.rs [hunk 2]
```diff
@@ -3031,3 +3033,201 @@ where
...
+    let browser_info: Option<BrowserInformation> = payment_attempt
+        .browser_info
+        .clone()
+        .map(|browser_information| browser_information.parse_value("BrowserInformation"))
+        .transpose()
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "browser_info",
+        })?;
+    let payment_connector_name = payment_attempt
+        .connector
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments.rs:3174]: use `merchant_connector_id`

File: crates/router/src/routes/payments.rs [hunk 3]
```diff
@@ -1208,6 +1208,58 @@ pub async fn payments_incremental_authorization(
...
+        payload,
+        |state, auth, req| {
+            payments::payment_external_authentication(
+                state,
+                auth.merchant_account,
+                auth.key_store,
+                req,
+            )
+        },
+        &auth::PublishableKeyAuth,
```
[the remark this answers is not in the document]
Comment (sai-harsha-vardhan) [reply crates/router/src/routes/payments.rs:1257]: as of now, it's not required

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]
No connector specific changes!

**Review by lsampras — APPROVED** [review]

**Review by ShankarSinghC — APPROVED** [review]

**Pull request merged**